File: GLSurfaceViewTest/src/main/java/com/lilong/glsurfaceviewtest/MainActivity.java
package com.lilong.glsurfaceviewtest;

import android.app.Activity;
import android.opengl.GLSurfaceView;
import android.os.Bundle;

public class MainActivity extends Activity {

    public static final String TAG = "GTest";

    private GLSurfaceView glSurfaceView;
    private GLSurfaceView.Renderer glSurfaceRender;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        glSurfaceView = findViewById(R.id.glSurfaceView);
        glSurfaceRender = new MainRender(getResources());
        // GLSurfaceView独有方法
        glSurfaceView.setRenderer(glSurfaceRender);
        glSurfaceView.setRenderMode(GLSurfaceView.RENDERMODE_CONTINUOUSLY);
    }

}

File: KaptTest/src/main/java/com/lilong/kapttest/MainActivity.kt
package com.lilong.kapttest

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.lilong.kapttest.feature.FeaturePresenter

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val presenter = FeaturePresenter()
        presenter.featureFunction()
    }
}
